src/components/IconicDrop/styled.ts

```diff
@@ -0,0 +1,36 @@
+import styled from 'styled-components';
+
+import { IconicDropProps } from './types';
+
+function getSize({ size, width = '45px', height = '45px' }: IconicDropProps): string {
+    let result = '';
+
+    if (size) {
+        return `
+        width: min(100%, ${size});
+        height: auto;
+    `;
+    }
+
+    if (width) {
+        result = `width: min(100%, ${width});`;
+    }
+
+    if (height) {
+        result = `height: ${height};`;
+    }
```

Comment:
Not sure if you want to append, rather than replace the `result` string.

```suggestion
    if (width) {
        result += `width: min(100%, ${width});`;
    }

    if (height) {
        result += `height: ${height};`;
    }
```
In case the user provides both width and height?